tests/test_prompt.py

```diff
@@ -95,22 +95,32 @@ def test_prompt_for_config(self, monkeypatch, context):
                 {'key': 'default'},
             ),
             ({'cookiecutter': {'key?{{False}}': 'default'}}, {'key': 'default'}),
+            (
+                {
+                    "cookiecutter": {
+                        "is_storage": "yes",
+                        "access_mode?{{cookiecutter.is_storage=='yes'}}": [
+                            "ReadWriteOnce",
+                            "ReadOnlyMany",
+                            "ReadWriteMany",
+                        ],
+                        "app": "app",
+                    }
+                },
+                {"is_storage": "yes", "access_mode": "ReadWriteOnce", "app": "app"},
+            ),
         ],
         ids=[
             'true dependent question',
             'false dependent question',
+            'true dependent question with variable',
         ],
     )
     def test_prompt_for_config_dependent_question(
         self, monkeypatch, context, expected_dict
     ):
         """Verify `prompt_for_config` call `read_user_variable` on text request."""
-        monkeypatch.setattr(
-            'cookieninja.prompt.read_user_variable',
-            lambda var, default: default,
-        )
-
-        cookiecutter_dict = prompt.prompt_for_config(context)
+        cookiecutter_dict = prompt.prompt_for_config(context, True)
```

Comment:
Defaulted parameters should usually be specified as keyword arguments. This way multiple parameters can be added, and without them all having to be specified.

Reply:
fixed